sonar-plugin-api/src/main/java/org/sonar/api/component/ResourcePerspectives.java

```diff
@@ -22,41 +22,14 @@
 import javax.annotation.CheckForNull;
 import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.batch.fs.InputPath;
-import org.sonar.api.issue.Issuable;
+import org.sonar.api.batch.sensor.SensorContext;
 import org.sonar.api.resources.Resource;
-import org.sonar.api.source.Highlightable;
-import org.sonar.api.source.Symbolizable;
-import org.sonar.api.test.TestPlan;
-import org.sonar.api.test.Testable;
 
 /**
- * Use this component to create perspective from resources or {@link InputPath}
- * Only on batch-side.
- * 
- * <pre>
- * public class MySensor implements Sensor {
- *   private final ResourcePerspectives perspectives;
- *
- *   public MySensor(ResourcePerspectives perspectives) {
- *     this.perspectives = perspectives;
- *   }
- *   
- *   public void analyse(Project module, SensorContext context) {
- *      // Get some Resource or InputFile/InputPath
- *      Highlightable highlightable = perspectives.as(Highlightable.class, inputPath);
- *      if (highlightable != null) {
- *        ...
- *      }
- *   }
- * }
- * </pre>
- * @see Issuable
- * @see Highlightable
- * @see Symbolizable
- * @see Testable
- * @see TestPlan
  * @since 3.5
+ * @deprecated since 5.6 all features are available on {@link SensorContext}
  */
+@Deprecated
 public interface ResourcePerspectives {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)